backend/src/routes/api/images/imageUtils.ts

```diff
@@ -194,16 +194,10 @@ const mapImageStreamToBYONImage = (is: ImageStream): BYONImage => ({
   description: is.metadata.annotations['opendatahub.io/notebook-image-desc'],
   visible: is.metadata.labels['opendatahub.io/notebook-image'] === 'true',
   error: getBYONImageErrorMessage(is),
-  packages:
-    is.spec.tags &&
-    (JSON.parse(
-      is.spec.tags[0].annotations['opendatahub.io/notebook-python-dependencies'],
-    ) as BYONImagePackage[]),
-  software:
-    is.spec.tags &&
-    (JSON.parse(
-      is.spec.tags[0].annotations['opendatahub.io/notebook-software'],
-    ) as BYONImagePackage[]),
+  packages: jsonParsePackage(
+    is.spec.tags?.[0]?.annotations['opendatahub.io/notebook-python-dependencies'],
+  ),
+  software: jsonParsePackage(is.spec.tags?.[0]?.annotations['opendatahub.io/notebook-software']),
```

Comment:
Do we not want to conditionally check to see if `annotations` is there?

Reply:
It might not really matter if undefined is being passed in anyways. 

`JSON.parse(undefined)` and `JSON.parse("invalid json")` both would be caught by the function and properly returned as an empty array

Reply:
I think Andrew worries about the possible type error that could crush the backend. I can update it to `is.spec.tags?.[0]?.annotations?.['opendatahub.io/notebook-software']`

Reply:
Actually, that hardly happens unless someone directly changes the YAML file of the image stream.

Reply:
why does this not produce a type error now? if the annotation can be undefined

```
  annotations?: { [key: string]: string };
```

and the function expects a non undefined string? Am i missing something

Reply:
Ohhhhh, yes, you are right... I already updated it, anyway. it will be refactored in the BYON phase 2

Reply:
ohhh right, good catch, kinda weird that didn't raise an error since the annotation was optional, first read I swear I thought it was optionally chained.